### ISSUE TITLE
Should the normalizer for classification loss consider background anchors?

### ISSUE THREAD

**Issue by zjysteven** [fizyr/keras-retinanet#1254]
Hi! Thank you for this implementation. This is not actually an issue, but I just want to confirm something. I just noticed that when computing the classification loss, you are normalizing it with only positive anchors as below.
https://github.com/fizyr/keras-retinanet/blob/a8bbd6535655602a84f194b1557be1ed5180fa71/keras_retinanet/losses.py#L61-L65

However, if I'm understanding it correctly, background anchors (whose anchor state is 0 and targets are all zeros) also contribute to the loss, right? So if this is the case, why don't we consider background anchors when computing the normalizer? 

I actually have a guess for this question, which is that the number of background anchors is so large than positive ones, and considering them will lead to a rather small loss (e.g. around 0.0020) which will harm the optimization. But I'm not sure and want some confirmation from you guys. Thanks!

**Comment by zjysteven** [comment]
Just noticed a similar question was raised in https://github.com/fizyr/keras-retinanet/issues/709. However I decide to leave this issue open for now for some possible discussions.

**Comment by hgaiser** [comment]
Hey, good question. As you found, it was indeed already discussed in #709. It doesn't make 100% sense to me, but as you guessed, normalizing by all anchors results in a very low loss value.

**Comment by zjysteven** [comment]
Thanks for the clarification! According to my experiments which I finished just now, considering background anchors will indeed lead to small loss values and eventually make the trained model's output probabilities all around 0. I think this confirmed my guess.
